crates/svm-ffi/src/byte_array.rs

```diff
@@ -0,0 +1,284 @@
+use std::convert::TryFrom;
+use std::string::FromUtf8Error;
+
+use byteorder::{BigEndian, ByteOrder};
+
+use svm_types::{WasmType, WasmValue};
+
+/// FFI representation for a byte-array
+///
+/// # Example
+///
+/// ```rust
+/// use std::convert::TryFrom;
+/// use std::string::FromUtf8Error;
+/// use svm_ffi::svm_byte_array;
+///
+/// let s1 = "Hello World!".to_string();
+/// let bytes: svm_byte_array = s1.into();
+///
+/// let s2 = String::try_from(bytes).unwrap();
+/// assert_eq!(s2, "Hello World!".to_string());
+/// ```
+///
+#[allow(non_camel_case_types)]
+#[derive(Clone)]
+#[repr(C)]
+pub struct svm_byte_array {
```

Comment:
Need to be exposed via the C API.

Reply:
yes - I'm on it

Reply:
The only workaround I've found for now is:
1. use `cbindgen` for the `svm-ffi` crate - it should output a C header named `svm_types.h`
This header file will contain the definition for `svm_byte_array` along with other used FFI-structs within `svm-runtime-c-api`

2. The `cbindgen` of `svm-runtime-c-api` will still output the `svm.h` header but it will have an include statement of `include "svm_types.h"` inside.

3. Each successful GitHub Actions CI run will have `svm_types.h` in its generated artifacts.

Reply:
I tested the artifacts - works fine.